File: leetcode/python/easy/climbing_stairs.py
class Solution:
    def __init__(self) -> None:
        self._cache = {}

    def climbStairs(self, n: int) -> int:
        if n in self._cache:
            return self._cache[n]
        if n <= 2:
            return n
        result = self.climbStairs(n - 1) + self.climbStairs(n - 2)
        self._cache[n] = result
        return result


print(Solution().climbStairs(2))
print(Solution().climbStairs(3))

File: leetcode/scala/easy/climbing_stairs/climbing_stairs.scala
object Solution {
    def climbStairs(n: Int): Int = {
        if(n <= 2) {return n}
        var a: Int = 1
        var b: Int = 2
        var steps: Int = 2
        while(steps < n){
            var c = a + b
            a = b
            b = c
            steps = steps + 1
        }
        b
    }
}
